task1/mapFunction.js

```diff
@@ -0,0 +1,54 @@
+const notes = [
+  {
+    id: 1,
+    title: "Recipe",
+    description: "Ingredients include 2 eggs...",
+    pagesCount: 2,
+    isMarked: false,
+    access: [],
+  },
+  {
+    id: 2,
+    title: "Workouts",
+    description: "3 sets of squats...",
+    pagesCount: 1,
+    isMarked: true,
+    access: [],
+  },
+  {
+    id: 3,
+    title: "Passwords",
+    description: "VISA ...",
+    pagesCount: 6,
+    isMarked: true,
+    access: [],
+  },
+  {
+    id: 4,
+    title: "To Do 2021",
+    description: "1. Learn JS...",
+    pagesCount: 3,
+    isMarked: false,
+    access: [],
+  },
+];
+
+Array.prototype.myMap = function myMap(func) {
+  if (!func) {
+    throw new Error("Function is not defined");
+  }
+  let result = [];
+  for (let i = 0; i < this.length; i++) {
+    result.push(func(this[i]));
+  }
+  return result;
+};
+
+console.log(
+  notes.myMap((elem) => {
+    let elemHash = {};
+    elemHash.id = elem.id;
+    elemHash.title = elem.title;
+    return elemHash;
+  })
```

Comment:
We can implement it easier and with renaming of the `elem`:
```
notes.myMap((note) => ({
  id: note.id,
  title:  note.title.
})
```
There is no need to create an additional object inside